src/AutoSaveStatus/useAutoSaveStatus.ts

```diff
@@ -0,0 +1,30 @@
+import { useContext, useEffect, useRef } from "react";
+import { AutoSaveStatus, AutoSaveStatusContext } from "./AutoSaveStatusProvider";
+
+/**
+ * Provides access to the nearest AutoSaveContext data, with an optional
+ * parameter to reset from "Done" to "Idle" after a given timeout
+ */
+export function useAutoSaveStatus(resetToIdleTimeout?: number) {
+  const autoSave = useContext(AutoSaveStatusContext);
+  const { status, resetStatus } = autoSave;
+  const resetToIdleTimeoutRef = useRef<number | null>(null);
+
+  /** Resets AutoSaveStatus from "Done" to "Idle" after a timeout, if one is provided */
+  useEffect(() => {
+    if (resetToIdleTimeout === undefined) return;
```

Comment:
**question:** Will this ever be `undefined`? Based on its type it should only ever be a number or `null`.

Reply:
This is the parameter, not the ref. Maybe a variable name update is in order.

Reply:
ahh, gotcha. I missed that.

Reply:
Can we default to a number instead then? Maybe 3s?

Reply:
So my overoptimization is this is going to get passed in via the indicator:

`<AutoSaveIndicator resetTimer={7_000} />` which is going to lean on `useAutoSaveStatus(resetTimer)` to potentially reset the status.

But `useFormState()` also calls this hook and it shouldn't have a timeout. So I'd prefer not to default it to anything, because `undefined` is a perfectly valid value here.

Reply:
Effectively, resetting from Done back to Idle/uncalled is an opt-in feature. If it's undefined, it's disabled by returning immediately. That's what this line toggles.

Reply:
Okay. In my opinion, I think we'd always want to reset back to the idle state. Persisting the "Done" state on the page until the user makes another change feels odd to me. I agree it was an over-optimization, but I am not sure its needed. I think I'd rather have us default to whatever standard timeout UX/Design has decided on. This will help ensure consistent timings across the different places we drop this indicator in. Giving the option to customize that timing might even be an over optimization at this point, and potentially lead to inconsistencies. Though, the option is good for others who may be using form state outside of Blueprint.